src/e2e-test/features/gcs/source/GCSSourceToBigQuery.feature

```diff
@@ -285,3 +285,268 @@ Feature: GCS source - Verification of GCS to BQ successful data transfer
+    Then Wait till pipeline preview is in running state
+    Then Open and capture pipeline preview logs
+    Then Verify the preview run status of pipeline in the logs is "succeeded"
+    Then Close the pipeline logs
+    Then Close the preview
+    Then Deploy the pipeline
+    Then Run the Pipeline in Runtime
+    Then Enter runtime argument value "gcsJsonFileSchema" for key "OutSchema"
+    Then Run the Pipeline in Runtime with runtime arguments
+    Then Wait till pipeline is in running state
+    Then Open and capture logs
+    Then Verify the pipeline status is "Succeeded"
+    Then Get count of no of records transferred to target BigQuery Table
+    Then Validate the values of records transferred from GCS bucket file is equal to the values of target BigQuery table
+
+  @GCS_CSV_TEST @BQ_SINK_TEST
+  Scenario: To verify Successful GCS to BigQuery data transfer with enable data file encryption flag true
+    Given Open Datafusion Project to configure pipeline
+    When Select plugin: "GCS" from the plugins list as: "Source"
+    When Expand Plugin group in the LHS plugins list: "Sink"
+    When Select plugin: "BigQuery" from the plugins list as: "Sink"
+    Then Connect source as "GCS" and sink as "BigQuery" to establish connection
+    Then Open GCS source properties
+    Then Enter GCS property projectId and reference name
+    Then Override Service account details if set in environment variables
+    Then Enter GCS source property path "gcsCsvFile"
+    Then Select GCS property format "csv"
+    Then Toggle GCS source property skip header to true
+    Then Validate output schema with expectedSchema "gcsCsvFileSchema"
+    Then Validate "GCS" plugin properties
+    Then Select radio button plugin property: "encrypted" with value: "true"
+    Then Close the GCS properties
+    Then Open BigQuery sink properties
+    Then Override Service account details if set in environment variables
+    Then Enter the BigQuery sink mandatory properties
+    Then Validate "BigQuery" plugin properties
+    Then Close the BigQuery properties
+    Then Save the pipeline
+    Then Preview and run the pipeline
+    Then Wait till pipeline preview is in running state
+    Then Open and capture pipeline preview logs
+    Then Verify the preview run status of pipeline in the logs is "succeeded"
+    Then Close the pipeline logs
+    Then Close the preview
+    Then Deploy the pipeline
+    Then Run the Pipeline in Runtime
+    Then Wait till pipeline is in running state
+    Then Open and capture logs
+    Then Verify the pipeline status is "Succeeded"
+    Then Get count of no of records transferred to target BigQuery Table
+    Then Validate the values of records transferred from GCS bucket file is equal to the values of target BigQuery table
+
+  @GCS_CSV_TEST @BQ_SINK_TEST @BigQuery_Sink_Required
+  Scenario:To verify successful records transfer from GCS source to BigQuery sink with macro fields enabled at source
```

Comment:
No of records transferred is `0` in this test, is it expected?

![image](https://github.com/user-attachments/assets/130ef161-65db-4683-90f6-3b87e91310b9)